[CI] TransformUpdaterTests testTransformUpdateRewrite failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/vepsfbiyzgeqc/tests/:x-pack:plugin:transform:test/org.elasticsearch.xpack.transform.action.TransformUpdaterTests/testTransformUpdateRewrite

**Reproduction line:**
`./gradlew ':x-pack:plugin:transform:test' --tests "org.elasticsearch.xpack.transform.action.TransformUpdaterTests.testTransformUpdateRewrite" -Dtests.seed=F9117854AFFB3321 -Dtests.locale=en-CA -Dtests.timezone=Australia/Yancowinna -Druntime.java=16`

**Applicable branches:**
master, 7.16, 8.0

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.transform.action.TransformUpdaterTests&tests.test=testTransformUpdateRewrite

**Failure excerpt:**
```
java.lang.AssertionError: expected:<{
  "id" : "n",
  "version" : "7.12.0",
  "create_time_string" : "2021-12-13T13:33:17.406Z",
  "create_time" : 1639402397406,
  "source" : {
    "index" : [
      "rmPpqIafKw",
      "zSumxHbIwY",
      "ISuqFvWUup",
      "mwlzmYLkcH",
      "GCPgudcNWy"
    ],
    "query" : {
      "match_all" : {
        "boost" : 1.0
      }
    },
    "runtime_mappings" : {
      "fT" : {
        "type" : "long"
      }
    }
  },
  "dest" : {
    "index" : "lzNlvVVhGa"
  },
  "sync" : {
    "time" : {
      "field" : "TtSbx",
      "delay" : "1786869864709054679ms"
    }
  },
  "latest" : {
    "unique_key" : [
      "yqHu",
      "RgCihRQ"
    ],
    "sort" : "xi"
  },
  "description" : "dAqUClMyRYCWRSTfZiQtSPxfxazAMzvGEveGhOjmPFGKDxxCkXobpjltsSIHpjrVGxpOTPJoFbKWRJkHJADbcZwAAsOtHFVUWcozGYQwhWzGQlgcqFtUVmmdcIBgjhBIjvVPiqqBQKEyBppiZbGBauBFwhTTHREtkFecjuJVpRLIitzAMQzRaUrJTFXsuofHnNLxPuOcJBdnuGBtDYaeUgAAscOYrrHgIcvliZWLGViYzyEHjovNLXoyhulBHAmfUgSaRAF",
  "settings" : {
    "docs_per_second" : 0.027091146,
    "use_point_in_time" : false
  },
  "retention_policy" : {
    "time" : {
      "field" : "QnZ",
      "max_age" : "359267681ms"
    }
  }
}> but was:<{
  "id" : "n",
  "version" : "7.16.1",
  "create_time_string" : "2021-12-13T13:33:17.288Z",
  "create_time" : 1639402397288,
  "source" : {
    "index" : [
      "iNCeNqVvch",
      "XxuqouCILR",
      "zdMModPZSj",
      "qXNZaSdHPw",
      "LSMFzRCcgU",
      "gEtfmKAzvL",
      "gpRiLKYJiA",
      "yJtdsXMlbm",
      "bghXkliMHK",
      "gEIElozieK"
    ],
    "query" : {
      "match_none" : {
        "boost" : 1.0
      }
    },
    "runtime_mappings" : {
      "FmAN" : {
        "type" : "keyword"
      },
      "UFFSTV" : {
        "type" : "keyword"
      }
    }
  },
  "dest" : {
    "index" : "NjLuScGrgK"
  },
  "frequency" : "2278143ms",
  "sync" : {
    "time" : {
      "field" : "IRIKeaf",
      "delay" : "3795441428505943410ms"
    }
  },
  "latest" : {
    "unique_key" : [
      "EV"
    ],
    "sort" : "qCGJlYTe"
  },
  "description" : "eEOodklHfmtdlejqHAhsBpJBgXigXReIVADuolJCaJBGuoGUABlolBdJzVKHfVzflzNbgcfWCHHMOhNVjppPnwnOXlVrzHKRDFwYugroazlVgQRhynPLTWvCNUesZoMeTAquJsGzlStHbHYFwGYdFrXnxSxYKNXbCktYxMlFEfoMOknbuDgpclMZKAWAoLkpUdOsibVrgMpFwPapHabGGEwqhyRHKknWNuhTzMUSOfwBWjuZSlMkzLpzaSQjndnmParsYzGfKhCqXaOgcSoFWjHYpHcaHlOaoBgviWFlwyDBFhXCIlsKZxWxoWvOSvxgdZLKkkwLHoqAtrKKxyIxEwQbxIEDvdfDxbcLzvRNdqArgAODqYJPNBQssUVobeTJEdDfXQExfMHnDsGIdWeyApUvDflXLUmgcbhIUTsyGFrTKHZagtIjzFhIVuXYLVCzpkmrfXIbSXUDpCUKkDWicgufJbUIsdczjtxNnZaOyeuvlEFDKcHEIoafeQQvabIwmkETQeNXMqGHolNibkXyemWrzpiOcsGkdhTJOIAX",
  "settings" : {
    "docs_per_second" : 0.1273936,
    "dates_as_epoch_millis" : true,
    "align_checkpoints" : false,
    "use_point_in_time" : true
  },
  "_meta" : {
    "VBgTBLZfHv" : null,
    "v" : "ft",
    "f" : null,
    "Kc" : 8864307294638466136,
    "JkUFdoBUEj" : null,
    "glGydWdgL" : "EryDkSUr",
    "AmNvn" : {
      "KDcv" : "fwtU"
    },
    "NOVQdeTZ" : null,
    "doAxsp" : null
  }
}>

  at __randomizedtesting.SeedInfo.seed([F9117854AFFB3321:A52A73E67655ACB4]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.failNotEquals(Assert.java:834)
  at org.junit.Assert.assertEquals(Assert.java:118)
  at org.junit.Assert.assertEquals(Assert.java:144)
  at org.elasticsearch.xpack.transform.action.TransformUpdaterTests.lambda$testTransformUpdateRewrite$37(TransformUpdaterTests.java:315)
  at org.elasticsearch.xpack.transform.action.TransformUpdaterTests.lambda$assertAsync$38(TransformUpdaterTests.java:350)
  at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:136)
  at org.elasticsearch.action.LatchedActionListener.onResponse(LatchedActionListener.java:30)
  at org.elasticsearch.xpack.transform.persistence.InMemoryTransformConfigManager.getTransformConfiguration(InMemoryTransformConfigManager.java:217)
  at org.elasticsearch.xpack.transform.action.TransformUpdaterTests.lambda$testTransformUpdateRewrite$36(TransformUpdaterTests.java:312)
  at org.elasticsearch.xpack.transform.action.TransformUpdaterTests.assertAsync(TransformUpdaterTests.java:353)
  at org.elasticsearch.xpack.transform.action.TransformUpdaterTests.assertConfiguration(TransformUpdaterTests.java:327)
  at org.elasticsearch.xpack.transform.action.TransformUpdaterTests.testTransformUpdateRewrite(TransformUpdaterTests.java:311)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:567)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:831)

```

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/action/TransformUpdaterTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/action/TransformUpdaterTests.java
@@ -260,8 +260,11 @@ public class TransformUpdaterTests extends ESTestCase {
                 assertEquals(stateDoc.getTransformStats(), storedDocAndVersion.v1().getTransformStats());
             }
         );
+    }
 
-        // same as dry run
+    public void testTransformUpdateDryRun() throws InterruptedException {
+        InMemoryTransformConfigManager transformConfigManager = new InMemoryTransformConfigManager();
+
         TransformConfig oldConfigForDryRunUpdate = TransformConfigTests.randomTransformConfig(
             randomAlphaOfLengthBetween(1, 10),
             VersionUtils.randomVersionBetween(
@@ -277,6 +280,7 @@ public class TransformUpdaterTests extends ESTestCase {
             config -> {}
         );
 
+        TransformConfigUpdate update = TransformConfigUpdate.EMPTY;
         assertUpdate(
             listener -> TransformUpdater.updateTransform(
                 securityContext,

